### PR TITLE
[chore] update to torch v1.8.0

### PR DESCRIPTION
**Pull request by msbaines**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by min-xu-ai — APPROVED** [review]
Awesome!!!

I vaguely remember there is a reference of torch version in docs or setup.py or something like that. Do a quick check there too?

File: fairscale/utils/testing.py [hunk 1]
```diff
@@ -146,12 +146,17 @@ def dist_init(rank: int, world_size: int, filename: str, filename_rpc: str = "")
 
         torch.distributed.init_process_group(backend=backend, rank=rank, world_size=world_size, init_method=url)
 
+        tp_options = {"init_method": url_rpc}
+        # Workaround for bug in torch v1.8.0. Should be fixed in v1.8.1
```
Comment (blefaudeux) [inline fairscale/utils/testing.py:150]: oh wow, thanks for catching that

**Review by blefaudeux — APPROVED** [review]
LGTM, many thanks @msbaines for tackling that

**Review by anj-s — APPROVED** [review]
Thank you for this change @msbaines!

**Pull request merged**